test/unit/specs/vislib/lib/y_axis.js

```diff
@@ -210,128 +225,132 @@ define(function (require) {
         });
       });
 
-      describe('should not return a nice scale when defaultYExtents is true', function () {
+      describe('validate user defined values', function () {
+        var newDomain;
+        var min;
+        var max;
+
         beforeEach(function () {
-          createData(defaultGraphData);
-          yAxis._attr.defaultYExtents = true;
-          yAxis.getYAxis(height);
-          yAxis.render();
+          yAxis._attr.mode = 'stacked';
+          yAxis._attr.setYExtents = false;
+          yAxis._attr.yAxis = {};
         });
 
-        it('not return a nice scale', function () {
-          var min = _.min(_.flatten(defaultGraphData));
-          var max = _.max(_.flatten(defaultGraphData));
-          var domain = yAxis.yAxis.scale().domain();
-          expect(domain[0]).to.be(min);
-          expect(domain[1]).to.be(max);
+        it('should throw a NaN error', function () {
+          min = 'Not a number';
+          max = 12;
+
+          expect(function () {
+            yAxis._validateUserExtents(min, max);
+          }).to.throwError();
+        });
+
+        it('should return a decimal value', function () {
+          yAxis._attr.mode = 'percentage';
+          yAxis._attr.setYExtents = true;
+          domain = [];
+          domain[0] = yAxis._attr.yAxis.min = 20;
+          domain[1] = yAxis._attr.yAxis.max = 80;
+          newDomain = yAxis._validateUserExtents(domain);
+
+          expect(newDomain[0]).to.be(domain[0] / 100);
+          expect(newDomain[1]).to.be(domain[1] / 100);
+        });
+
+        it('should return the user defined value', function () {
+          domain = [20, 50];
+          newDomain = yAxis._validateUserExtents(domain);
+
+          expect(newDomain[0]).to.be(domain[0]);
+          expect(newDomain[1]).to.be(domain[1]);
+        });
+      });
+
+      describe('should throw an error when', function () {
+        var min;
+        var max;
+
+        it('min === max', function () {
+          min = 12;
+          max = 12;
+
+          expect(function () {
+            yAxis._validateAxisExtents(min, max);
+          }).to.throwError();
+        });
+
+        it('min > max', function () {
+          min = 30;
+          max = 10;
+
+          expect(function () {
+            yAxis._validateAxisExtents(min, max);
+          }).to.throwError();
         });
       });
     });
 
+    //describe('formatAxisLabel method', function () {
```

Comment:
Remove this commented out code. We can always fish it out of git if we need it back.